hashids_test.go

```diff
@@ -200,95 +201,127 @@ func TestAllocationsPerEncodeTypical(t *testing.T) {
 	minNumbers := []int64{0, 0, 0, 0}
 	mixNubers := []int64{math.MaxInt64, 0, 1024, math.MaxInt64 / 2}
 
-	checkAllocations(t, hid, singleNumber, 5)
+	checkAllocations(t, hid, singleNumber, 9)
```

Comment:
This one case increases from 5 -> 9 allocations. Could possibly reduce that but was going to just call it good enough for now, users who want the best performance should use HashWorkspace which only has a single allocation per encode (allocating the final string)